app/models/goal.rb

```diff
@@ -0,0 +1,60 @@
+class Goal < ActiveRecord::Base
+  belongs_to :setter, class_name: "User"
+  belongs_to :tender, class_name: "User"
+  belongs_to :charity
+  has_many :milestones
+  has_many :messages
+  has_many :pledges
+
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :tender, presence: true
+  validates :setter_pledge, presence: true
+
+  delegate :username, to: :setter, prefix: true
+
+  def announcement
+    Message.create(
+      user_id: 1,
```

Comment:
self.messages.create(user: User.system_user, content: content: "#{self.setter.username} has set a goal to '#{self.title}'.")